clojure-ts-mode.el

```diff
@@ -2032,6 +2055,135 @@ value is `clojure-ts-thread-all-but-last'."
+  (when-let* ((parent (treesit-node-parent node)))
+    (and (clojure-ts--list-node-p node)
+         (clojure-ts--vec-node-p parent)
+         (let ((grandparent (treesit-node-parent parent)))
+           (string= (clojure-ts--list-node-sym-text grandparent)
+                    "letfn")))))
+
+(defun clojure-ts--proxy-defn-p (node)
+  "Return non-nil if NODE is a function definition in a proxy form."
+  (when-let* ((parent (treesit-node-parent node)))
+    (and (clojure-ts--list-node-p node)
+         (string= (clojure-ts--list-node-sym-text parent) "proxy"))))
+
+(defun clojure-ts--defprotocol-defn-p (node)
+  "Return non-nil if NODE is a function definition in a defprotocol form."
+  (when-let* ((parent (treesit-node-parent node)))
+    (and (clojure-ts--list-node-p node)
+         (string= (clojure-ts--list-node-sym-text parent) "defprotocol"))))
+
+(defun clojure-ts--reify-defn-p (node)
+  "Return non-nil if NODE is a function definition in a reify form."
+  (when-let* ((parent (treesit-node-parent node)))
+    (and (clojure-ts--list-node-p node)
+         (string= (clojure-ts--list-node-sym-text parent) "reify"))))
+
+(defun clojure-ts-add-arity ()
+  "Add an arity to a function."
+  (interactive)
+  (if-let* ((sym-regex (rx bol
+                           (or "defn"
+                               "letfn"
+                               "fn"
+                               "defmacro"
+                               "defmethod"
+                               "defprotocol"
+                               "reify"
+                               "proxy")
+                           eol))
+            (parent-def-node (clojure-ts--search-list-form-at-point sym-regex))
+            (parent-def-sym (clojure-ts--list-node-sym-text parent-def-node))
+            (fn-node (cond
+                      ((string= parent-def-sym "letfn")
+                       (clojure-ts--parent-until #'clojure-ts--letfn-defn-p))
+                      ((string= parent-def-sym "proxy")
+                       (clojure-ts--parent-until #'clojure-ts--proxy-defn-p))
+                      ((string= parent-def-sym "defprotocol")
+                       (clojure-ts--parent-until #'clojure-ts--defprotocol-defn-p))
+                      ((string= parent-def-sym "reify")
+                       (clojure-ts--parent-until #'clojure-ts--reify-defn-p))
+                      (t parent-def-node))))
+      (let ((beg-marker (copy-marker (treesit-node-start parent-def-node)))
+            (end-marker (copy-marker (treesit-node-end parent-def-node))))
+        (cond
+         ((string= parent-def-sym "defprotocol")
+          (clojure-ts--add-arity-defprotocol-internal fn-node))
+         ((string= parent-def-sym "reify")
+          (clojure-ts--add-arity-reify-internal fn-node))
+         (t (clojure-ts--add-arity-internal fn-node)))
+        (indent-region beg-marker end-marker))
+    (user-error "No function at point")))
```

Comment:
That's not very precise, but I'm not sure what would be better (and still short) wording.

Reply:
Maybe "No suitable form to add an arity at point"?

Reply:
Sounds good.